src/tools/rbd/action/ImageMeta.cc

```diff
@@ -94,6 +94,24 @@ static int do_metadata_set(librbd::Image& image, const char *key,
   return r;
 }
 
+static int do_matadata_keycheck(librbd::Image& image, const char *key){
```

Comment:
This logic should be moved into librbd so that the API can return ```-ENOENT```. The place to put the similar logic is about here [1].

[1] https://github.com/ceph/ceph/blob/master/src/librbd/Operations.cc#L1430